### PR TITLE
[CI] Fix Connection time out in Python workflow

### PR DESCRIPTION
**Pull request by furqaankhan**
## Did you read the Contributor Guide?

- Yes, I have read the [Contributor Rules](https://sedona.apache.org/latest-snapshot/community/rule/) and [Contributor Development Guide](https://sedona.apache.org/latest-snapshot/community/develop/)

## Is this PR related to a JIRA ticket?

- No, this is a CI update. The PR name follows the format `[CI] my subject`

## What changes were proposed in this PR?

- Remove fetching spark jars from archive.apache.org due to download limits https://infra.apache.org/infra-ban.html.
- Restructure the workflow to remove unnecessary code.

## How was this patch tested?

- Passed CI

## Did this PR include necessary documentation updates?

- No, this PR does not affect any public API so no need to change the documentation.

### CODE REVIEW

File: .github/workflows/python.yml [hunk 1]
```diff
@@ -141,10 +129,25 @@ jobs:
...
+          wget --retry-connrefused --waitretry=10 --read-timeout=20 --timeout=15 --tries=5 https://repo.osgeo.org/repository/release/javax/media/jai_codec/${JAI_CODEC_VERSION}/jai_codec-${JAI_CODEC_VERSION}.jar
+          wget --retry-connrefused --waitretry=10 --read-timeout=20 --timeout=15 --tries=5 https://repo.osgeo.org/repository/release/javax/media/jai_imageio/${JAI_IMAGEIO_VERSION}/jai_imageio-${JAI_IMAGEIO_VERSION}.jar
+          mv -v jai_core-${JAI_CORE_VERSION}.jar /home/runner/.local/share/virtualenvs/python-${PYTHON_VERSION}/lib/python${PYTHON_VERSION}/site-packages/pyspark/jars
+          mv -v jai_codec-${JAI_CODEC_VERSION}.jar /home/runner/.local/share/virtualenvs/python-${PYTHON_VERSION}/lib/python${PYTHON_VERSION}/site-packages/pyspark/jars
+          mv -v jai_imageio-${JAI_IMAGEIO_VERSION}.jar /home/runner/.local/share/virtualenvs/python-${PYTHON_VERSION}/lib/python${PYTHON_VERSION}/site-packages/pyspark/jars
+      - name: Setup tmate session
+        uses: mxschmitt/action-tmate@v3.19
```
Comment (jbampton) [inline .github/workflows/python.yml:150]: https://infra.apache.org/github-actions-policy.html

Under External actions:

"You MUST pin all external actions to the specific git hash (SHA1) of the action that has been reviewed for use by the project. For instance, you MUST pin foobar/baz-action@8843d7f92416211de9ebb963ff4ce28125932878."

Does this apply here ?

Comment (furqaankhan) [reply .github/workflows/python.yml:150]: The tmate action won't be merged. I am using it to debug and will remove it when I merge.

**Comment by furqaankhan** [comment]
@jbampton @jiayuasu Not sure why the lint is not passing. It is passing on my fork's PR: https://github.com/furqaankhan/sedona/pull/2

**Comment by jbampton** [comment]
Hey @furqaankhan the `requirements txt fixer hook` is failing.  

Basically `docker/sedona-spark-jupyterlab/requirements.txt` is not sorted.

I have the fix in #1654

File: .github/workflows/python.yml [hunk 2]
```diff
@@ -141,32 +116,44 @@ jobs:
             echo "Patching Pipfile to use Shapely 1.x"
             sed -i 's/^shapely.*$/shapely="<2.0.0"/g' Pipfile
           fi
+          export PIPENV_CUSTOM_VENV_NAME=python-${PYTHON_VERSION}
           pipenv --python ${PYTHON_VERSION}
           pipenv install pyspark==${SPARK_VERSION}
           pipenv install --dev
           pipenv graph
+          pipenv run pip show pyspark
```
Comment (jiayuasu) [inline .github/workflows/python.yml:124]:
```suggestion
```

**Review by jiayuasu — CHANGES_REQUESTED** [review]
This value `/home/runner/.local/share/virtualenvs/python-${PYTHON_VERSION}/lib/python${PYTHON_VERSION}/site-packages/pyspark` is so long. Can you make it a global env?

**Comment by jiayuasu** [comment]
A quick example about global env from ChatGPT

```
name: Matrix CI Workflow

on: [push, pull_request]

jobs:
  build:
    runs-on: ubuntu-latest

    # Define a matrix of input values
    strategy:
      matrix:
        environment: [staging, production]
        node: [16, 18]

    # Use the matrix value inside `env`
    env:
      ENV_NAME: ${{ matrix.environment }}
      NODE_VERSION: ${{ matrix.node }}

    steps:
      - name: Check out repository
        uses: actions/checkout@v3

      - name: Print Matrix Inputs as Environment Variables
        run: |
          echo "Environment: $ENV_NAME"
          echo "Node Version: $NODE_VERSION"

      - name: Use Environment-Specific Logic
        run: |
          if [ "$ENV_NAME" == "production" ]; then
            echo "This is a production build.";
          else
            echo "This is a staging build.";
          fi
```

**Review by jiayuasu — APPROVED** [review]

**Comment by jiayuasu** [comment]
Great work!

**Pull request merged**